src/main/java/ch/njol/skript/effects/EffKnockback.java

```diff
@@ -0,0 +1,92 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.effects;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.RequiredPlugins;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.util.Direction;
+import ch.njol.util.Kleenean;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.event.Event;
+import org.bukkit.util.Vector;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Knockback")
+@Description("Apply the same velocity as a knockback to living entities in a direction. Mechanics such as knockback resistance will be factored in.")
+@Examples({
+	"knockback player north",
+	"knock victim (vector from attacker to victim) with strength 10"
+})
+@Since("INSERT VERSION")
+@RequiredPlugins("Paper 1.19.2+")
+public class EffKnockback extends Effect {
+
+	static {
+		if (Skript.methodExists(LivingEntity.class, "knockback", double.class, double.class, double.class))
+			Skript.registerEffect(EffKnockback.class, "(apply knockback to|knock[back]) %livingentities% [%direction%] [with (strength|force) %-number%]");
+	}
+
+	@SuppressWarnings("null")
+	private Expression<LivingEntity> entities;
+	@SuppressWarnings("null")
+	private Expression<Direction> direction;
+	@Nullable
+	private Expression<Number> strength;
+
+	@Override
+	@SuppressWarnings({"unchecked", "null"})
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		entities = (Expression<LivingEntity>) exprs[0];
+		direction = (Expression<Direction>) exprs[1];
+		strength = (Expression<Number>) exprs[2];
+		return true;
+	}
+
+	@Override
+	protected void execute(Event event) {
+		Direction direction = this.direction.getSingle(event);
+		if (direction == null)
+			return;
+
+		double strength = this.strength.getOptionalSingle(event).orElse(1).doubleValue();
```

Comment:
if `strength` is null, it'll throw an NPE, you should keep it to how it was before